### PR TITLE
Upgrade dependencies post 2.32.2 release

### PR DESCRIPTION
**Pull request by jcfilben**
#### What does this PR do?
Upgrades dependencies to the latest versions after 2.32.2 release.
Some typescript changes required adjustments to a typescript story and a typescript test

Snapshot changes are expected, coming from `grommet-icons v4.11.0`: https://github.com/grommet/grommet-icons/pull/312

Freezing `@testing-library/dom` at 9.3.0. `9.3.1` was introducing many errors related to wrapping tests in `act`. To be investigated/addressed separately.

#### What testing has been done on this PR?
storybook, e2e, build, jest
#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### Do the grommet docs need to be updated?
no
#### Should this PR be mentioned in the release notes?
no
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: src/js/all/stories/CustomThemed/ExtendTheme.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ const custom: ThemeType = {
   },
   anchor: {
     extend: (props) => css`
-      color: ${props.href ? 'green' : 'red'};
+      color: ${props['href'] ? 'green' : 'red'};
```
Comment (halocline) [inline src/js/all/stories/CustomThemed/ExtendTheme.tsx:16]: Assuming this was the linter. Did we have a rule change on us?

Comment (jcfilben) [reply src/js/all/stories/CustomThemed/ExtendTheme.tsx:16]: This didn't come from upgrading any lint dependencies. This change came from upgrading typescript.

Comment (jcfilben) [reply src/js/all/stories/CustomThemed/ExtendTheme.tsx:16]: After upgrading I got the error `TS2339: Property 'href' does not exist on type 'never'.`

**Review by halocline — APPROVED** [review]

**Review by taysea — APPROVED** [review]

**Comment by jcfilben** [comment]
Looks good, thanks @taysea!

**Pull request merged**